go-controller/pkg/ovn/loadbalancer/lb_cache.go

```diff
@@ -93,6 +100,50 @@ func (c *LBCache) removeVips(toRemove []DeleteVIPEntry) {
 	}
 }
 
+// RemoveSwitches removes the provided toRemoveSwitches which is a list
+// of switches from all the lb's.Switches in the LBCache.
+func (c *LBCache) RemoveSwitches(toRemoveSwitches []string) {
+	c.Lock()
+	defer c.Unlock()
+	for _, sw := range toRemoveSwitches {
+		switches, err := findTableLBs("logical_switch", sw)
+		if err != nil {
+			// best effort, we don't fail
+			klog.Error("Unable to remove load balancers from the switch %s", sw)
+		}
+
+		for switchname, lbuuids := range switches {
+			for _, lbuuid := range lbuuids {
+				if lb, ok := c.existing[lbuuid]; ok {
+					lb.Switches.Delete(switchname)
+				}
+			}
+		}
+	}
+}
+
+// RemoveSwitches removes the provided toRemoveRouters which is a list
+// of routers from all the lb's.Routers in the LBCache.
```

Comment:
nit:  dot between words lb and router.

Reply:
its intentional, the field is lb.Routers, I added the apostrophe for plural